src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -156,13 +173,45 @@ assign(SyntheticEvent.prototype, {
    * `PooledClass` looks for `destructor` on each instance it releases.
    */
   destructor: function() {
+    var setNull = setToNullOrWarning.bind(this);
     var Interface = this.constructor.Interface;
-    for (var propName in Interface) {
-      this[propName] = null;
+    for (var interfacePropName in Interface) {
+      setNull(interfacePropName);
+    }
+    otherNullableProps.forEach(setNull);
+    this._destructored = true;
+
+    function setToNullOrWarning(propName) {
+      if (__DEV__) {
+        Object.defineProperty(this, propName, {
+          set: function(val) {
+            // no-op
+            var warningCondition = false;
+            warning(
+              warningCondition,
+              'This synthetic event is reused for performance reasons. If you\'re ' +
+              'seeing this, you\'re setting property `' + propName + '` on a ' +
```

Comment:
I'm getting a linting error:

```
190:13  error  The second argument to warning must be a string literal  react-internal/warning-and-invariant-args
```

I think it's because of this line. Is there a reason I can't provide the `propName` here? I feel like it would be useful to have it.